templates/docs/base/forms.md

```diff
@@ -38,38 +38,80 @@ Note: The attribute `readonly` disables the input but it still retains a default
 
 ### Checkbox
 
-Use checkboxes to select one or more options, default checkboxes can appear in three states: selected, unselected and disabled.
+<span class="p-label--new">New</span>
+
+Use the checkbox component to select one or more options. To provide fully featured Vanilla style and behaviour of the checkbox a specific markup structure is needed around the checkbox input (see example below).
+
+To disable the checkbox component a `disabled` property needs to be added on respective `<input type='checkbox'>` element and `.is-disabled` class needs to be used on the component root `.p-checkbox` element.
+
+<div class="embedded-example"><a href="/docs/examples/patterns/forms/checkbox/" class="js-example">
+View example of the checkbox components
+</a></div>
+
+#### Inline checkbox
 
-<div class="embedded-example"><a href="/docs/examples/base/forms/checkboxes/" class="js-example">
-View example of the base checkboxes
+To use checkbox component in padded containers (table cells, some list items) use the `.p-checkbox--inline` variant of the component to make sure the checkbox and the label text are properly aligned with other inline text.
+
+<div class="embedded-example"><a href="/docs/examples/patterns/forms/checkbox-inline/" class="js-example">
+View example of the inline checkbox components
 </a></div>
 
-<span class="p-label--updated">Updated</span>
+#### Heading checkbox
 
-By default, checkboxes are vertically aligned to the baseline of text wrapped in a `label`, `h5`, `h6`, or `p` tag. If you need to align them to other elements, use one of the following classes:
-`is-h1`, `is-h2`, `is-h3`, `is-h4`, `is-h5`, `is-muted-heading`, `is-muted-inline-heading`, `is-inline-label`, or `is-table-header`.
+To use checkbox component in headings use the `.p-checkbox--heading` variant of the component to make sure the checkbox and the label text are properly aligned.
```

Comment:
```suggestion
To style the checkbox label as a heading, use the `.p-checkbox--heading` variant of the component. It ensures correct vertically alignement of the checkbox and the heading.
```

Reply:
`--heading` variant doesn't style the label as heading, only make sure checkbox is aligned, so skipping this.